### PR TITLE
chore: china partition support

### LINKED ISSUE #859: China regions(partition 'aws-cn') support
Create a project like below,

```bash
ecs-preview project init
Looks like you are using a workspace that's registered to project nexus3.
We'll use that as your project.
✘ Failed to create the infrastructure to manage container repositories under project nexus3.
Error: failed to create stack nexus3-infrastructure-roles: ResourceNotReady: failed waiting for successful resource state
```

Checked the events of cloudformation stack, see below error,
```
Partition "aws" is not valid for resource "arn:aws:iam::*:role/nexus3-adminrole". (Service: AmazonIdentityManagement; Status Code: 400; Error Code: MalformedPolicyDocument; Request ID: d902d514-01f0-4eb5-a341-2292ec5c95d5)
```

### PR DESCRIPTION
**Pull request by iamhopaul123**
fixes #859 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: internal/pkg/deploy/cloudformation/stack/app.go [hunk 1]
```diff
@@ -192,9 +193,12 @@ func (c *AppStackConfig) stackSetAdminRoleName() string {
...
 // StackSet.
-func (c *AppStackConfig) StackSetAdminRoleARN() string {
-	//TODO find a partition-neutral way to construct this ARN
-	return fmt.Sprintf("arn:aws:iam::%s:role/%s", c.AccountID, c.stackSetAdminRoleName())
+func (c *AppStackConfig) StackSetAdminRoleARN(region string) (string, error) {
+	partition, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
+	if !ok {
+		return "", fmt.Errorf("cannot find the partition for region %s", region)
+	}
+	return fmt.Sprintf("arn:%s:iam::%s:role/%s", partition.ID(), c.AccountID, c.stackSetAdminRoleName()), nil
```
Comment (huanjani) [inline internal/pkg/deploy/cloudformation/stack/app.go]: Q: would it be beneficial to make this a fmt var?

**Review by huanjani — APPROVED** [review]
Wow, these changes are going to have a huge, exciting impact! And speedy turnaround, too!

File: internal/pkg/aws/identity/identity_test.go [hunk 2]
```diff
@@ -32,6 +33,16 @@ func TestIdentity_Get(t *testing.T) {
 			},
 			wantErr: fmt.Errorf("get caller identity: %w", mockError),
 		},
+		"should return wrapped error if cannot parse the account arn": {
```
Comment (efekarakus) [inline internal/pkg/aws/identity/identity_test.go:37]: should we add a test that returns a china partition arn in `out.Arn`?

File: internal/pkg/deploy/cloudformation/stack/app.go [hunk 3]
```diff
@@ -192,9 +193,12 @@ func (c *AppStackConfig) stackSetAdminRoleName() string {
 
 // StackSetAdminRoleARN returns the role ARN of the role used to administer the Application
 // StackSet.
-func (c *AppStackConfig) StackSetAdminRoleARN() string {
-	//TODO find a partition-neutral way to construct this ARN
-	return fmt.Sprintf("arn:aws:iam::%s:role/%s", c.AccountID, c.stackSetAdminRoleName())
+func (c *AppStackConfig) StackSetAdminRoleARN(region string) (string, error) {
+	partition, ok := endpoints.PartitionForRegion(endpoints.DefaultPartitions(), region)
+	if !ok {
+		return "", fmt.Errorf("cannot find the partition for region %s", region)
```
Comment (efekarakus) [inline internal/pkg/deploy/cloudformation/stack/app.go]: nit
```suggestion
		return "", fmt.Errorf("find the partition for region %s", region)
```

File: internal/pkg/stream/ecs.go [hunk 4]
```diff
@@ -52,6 +53,8 @@ func (d ECSDeployment) done() bool {
 		return true
 	case rollOutCompleted:
 		return d.DesiredCount == d.RunningCount
+	case rollOutEmpty:
+		return d.DesiredCount == d.RunningCount
```
Comment (efekarakus) [inline internal/pkg/stream/ecs.go]: nit:

```suggestion
	case rollOutCompleted, rollOutEmpty:
		return d.DesiredCount == d.RunningCount
```

**Review by efekarakus — APPROVED** [review]
Feel free to remove label once addressed

**Comment by iamhopaul123** [comment]
> Feel free to remove label once addressed

They can't deploy circuit breaker to the China partition due to the Prime Day until next week. I'll merge it once they finish their deployment.

**Pull request merged**